ndn_hydra/client/functions/query.py

```diff
@@ -32,12 +32,13 @@ async def send_query(self, query: Name, node_name: str=None) -> None:
       Form a certain query and request that info from a node.
       """
       if not node_name:
-          named_query = self.repo_prefix + [Component.from_str("query")] + query
+        named_query = self.repo_prefix + [Component.from_str("query")]
       else:
-          named_query = self.repo_prefix + [Component.from_str("node")] + [Component.from_str(node_name)] + [Component.from_str("query")] + query
-
+        named_query = self.repo_prefix + [Component.from_str("node")] + [Component.from_str(node_name)] + [Component.from_str("query")]
+      app_param = query[0]
+      
       try:
-          data_name, meta_info, content = await self.app.express_interest(named_query, can_be_prefix=True, must_be_fresh=True, lifetime=3000)
+          data_name, meta_info, content = await self.app.express_interest(named_query, app_param, can_be_prefix=True, must_be_fresh=True, lifetime=3000)

```

Comment:
So the app_params here should be just the query (with indexing to 0). Say for instance, the client wants to query "/prefix/genomics/files", right now, you are just sending /prefix.

You can do the same thing that you did for the query handle.

Reply:
I'm not sure I understand. Do you want me to just change line 41 to:

data_name, meta_info, content = await self.app.express_interest(named_query, query[0], ...

and remove the variable?

Reply:
Not exactly. If you try to do a "/prefix/genomics" query, it will fail.

This is because you are indexing the query "[0]", technically you will only be sending /prefix.  To tell you exactly, remove line 38 and instead of "app_param" do "query".

Reply:
Gotcha, I must have misread. That makes sense to me now; I think I brain farted while reading. Fixed in most recent commit.